src/shared/components/splash-screen/SplashScreen.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import RNBootSplash from 'react-native-bootsplash';
+
+const SplashScreen = ({ hide, transitionDelay, children }) => {
+    const autoHide = hide === undefined;
+
+    useEffect(() => {
+        (autoHide || hide) && RNBootSplash.hide({ duration: transitionDelay });
```

Comment:
`RNBootSplash.hide()` is called more than once if `hide` changes. We should defend the component against such scenario for correctness.